core/src/apps/cardano/helpers/address_credential_policy.py

```diff
@@ -0,0 +1,197 @@
+from trezor.enums import CardanoAddressType, CardanoTxSigningMode
+
+from ..address import get_public_key_hash
+from .paths import CHAIN_STAKING_KEY, SCHEMA_PAYMENT, SCHEMA_STAKING
+from .utils import to_account_path
+
+if False:
+    from trezor.messages import CardanoAddressParametersType
+    from ..seed import Keychain
+
+
+CREDENTIAL_POLICY_SHOW = 0
+CREDENTIAL_POLICY_SHOW_KEY_PATH = 1 << 1
+CREDENTIAL_POLICY_SHOW_CHANGE = 1 << 2
+CREDENTIAL_POLICY_SHOW_OUTPUT = 1 << 3
+CREDENTIAL_POLICY_SHOW_ADDRESS = 1 << 4
+CREDENTIAL_POLICY_WARN_SCRIPT = 1 << 5
+CREDENTIAL_POLICY_WARN_POINTER = 1 << 6
+CREDENTIAL_POLICY_WARN_REWARD_ADDRESS = 1 << 7
+CREDENTIAL_POLICY_WARN_NO_STAKING = 1 << 8
+CREDENTIAL_POLICY_FAIL_OR_WARN_UNUSUAL = 1 << 9
+CREDENTIAL_POLICY_WARN_MISMATCH = 1 << 10
+
+CREDENTIAL_POLICIES_WARN = (
+    CREDENTIAL_POLICY_WARN_SCRIPT,
+    CREDENTIAL_POLICY_WARN_POINTER,
+    CREDENTIAL_POLICY_WARN_REWARD_ADDRESS,
+    CREDENTIAL_POLICY_WARN_NO_STAKING,
+    CREDENTIAL_POLICY_FAIL_OR_WARN_UNUSUAL,
+    CREDENTIAL_POLICY_WARN_MISMATCH,
+)
+
+ADDRESS_POLICY_SHOW_SPLIT = 0
+ADDRESS_POLICY_SHOW_SIMPLE = 1
+
+
+def get_address_policy(
+    keychain: Keychain, address_parameters: CardanoAddressParametersType
+) -> int:
+    if (
+        address_parameters.address_type == CardanoAddressType.BASE
+        and SCHEMA_PAYMENT.match(address_parameters.address_n)
+        and _do_base_address_credentials_match(
+            keychain,
+            address_parameters.address_n,
+            address_parameters.address_n_staking,
+            address_parameters.staking_key_hash,
+        )
+    ):
+        return ADDRESS_POLICY_SHOW_SIMPLE
```

Comment:
Instead of :
![image](https://user-images.githubusercontent.com/7520735/129472086-4a884753-04ac-4536-9057-32f21dc0b1e2.png)
just this is shown:
![image](https://user-images.githubusercontent.com/7520735/129472094-740c6a34-9aa3-46f5-bf23-342a99783015.png)
I think it's worth it as that is perhaps 90% of the cases.